genesis/process/intermediate/txExecutionProcessor.go

```diff
@@ -0,0 +1,104 @@
+package intermediate
+
+import (
+	"math"
+	"math/big"
+
+	"github.com/ElrondNetwork/elrond-go/core/check"
+	"github.com/ElrondNetwork/elrond-go/data/state"
+	transactionData "github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/genesis"
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+type txExecutionProcessor struct {
+	txProcessor process.TransactionProcessor
+	accounts    state.AccountsAdapter
+}
+
+// NewTxExecutionProcessor is able to execute a transaction
+func NewTxExecutionProcessor(
+	txProcessor process.TransactionProcessor,
+	accounts state.AccountsAdapter,
+) (*txExecutionProcessor, error) {
+	if check.IfNil(txProcessor) {
+		return nil, process.ErrNilTxProcessor
+	}
+	if check.IfNil(accounts) {
+		return nil, process.ErrNilAccountsAdapter
+	}
+
+	return &txExecutionProcessor{
+		txProcessor: txProcessor,
+		accounts:    accounts,
+	}, nil
+}
+
+// ExecuteTransaction will try to assemble a transaction and execute it against the accounts db
+func (tep *txExecutionProcessor) ExecuteTransaction(
```

Comment:
I do not think this is component is needed. maybe it is too fragmented.

Reply:
It will help testing the deploy and delegation processors. That is why it was embedded, contains all functions regarding assembling & processing the transaction + helper functions. It will be easily mocked inside genesis package.

Reply:
it is still to fragmented I think. but can stay.